src/nvim/api/tabpage.c

```diff
@@ -95,6 +199,35 @@ void nvim_tabpage_del_var(Tabpage tabpage, String name, Error *err)
   dict_set_var(tab->tp_vars, name, NIL, true, false, err);
 }
 
+/// Sets the current window in a tabpage
+///
+/// @param tabpage  Tabpage handle, or 0 for current tabpage
+/// @param win Window handle
+/// @param[out] err Error details, if any
+/// @return Window handle
+void nvim_tabpage_set_win(Tabpage tabpage, Window win, Error *err)
+  FUNC_API_SINCE(12)
+{
+  tabpage_T *tab = find_tab_by_handle(tabpage, err);
+
+  if (!tab) {
+    return;
+  }
+
+  if (tab == curtab) {
+    nvim_set_current_win(win, err);
+    return;
+  }
+
+  win_T *wp = find_window_by_handle(win, err);
+
+  if (!wp) {
+    return;
+  }
+
+  tab->tp_curwin = wp;
```

Comment:
what happens if `win` belonged to another tab page? needs to check for this and throw an error, I think. (could be done before the `curtab` special case, otherwise that function would switch the current tab which seems confusing).

Reply:
Right, thanks! Just pushed a change addressing this :)